bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/JSONParser.java

```diff
@@ -874,28 +875,49 @@ private void processNonStringValue(ValueType type) throws JsonParserException {
                     }
                 } else {
                     try {
-                        long longValue = Long.parseLong(str);
-                        switch (type) {
-                            case ARRAY_ELEMENT:
-                                ((ArrayValue) this.currentJsonNode).append(longValue);
-                                break;
-                            case FIELD:
-                                ((MapValueImpl<BString, Object>) this.currentJsonNode).put(
-                                        StringUtils.fromString(this.fieldNames.pop()), longValue);
-                                break;
-                            case VALUE:
-                                currentJsonNode = longValue;
-                                break;
-                            default:
-                                break;
+                        if (ch == '-' && !fromJsonFloatStringMode && !fromJsonDecimalStringMode && isZero(str)) {
```

Comment:
We don't need `!fromJsonFloatStringMode` in first if condition due to second `else if` condition.